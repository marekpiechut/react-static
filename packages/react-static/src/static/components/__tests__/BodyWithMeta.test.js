/**
 * @jest-environment jsdom
 */
import React from 'react'
import makeBodyWithMeta from '../BodyWithMeta'
import renderer from 'react-test-renderer'

describe('BodyWithMeta', () => {
  test('when route is a static route', async () => {
    process.env.REACT_STATIC_ASSETS_PATH = 'assets/path'
    const BodyWithMeta = await makeBodyWithMeta({
      head: { bodyProps: { lang: 'en' } },
      route: { redirect: false },
      inlineScripts: {
        routeInfo: {
          script: 'script',
        },
      },
      clientScripts: ['main.js', 'bootstrap.js'],
      plugins: [],
    })

    const bodyWithMeta = renderer
      .create(
        <BodyWithMeta className="body">
          <div>static page</div>
        </BodyWithMeta>
      )
      .toJSON()

    expect(bodyWithMeta).toMatchSnapshot()
  })

  test('when route is a redirect route', async () => {
    const BodyWithMeta = await makeBodyWithMeta({
      head: { bodyProps: { lang: 'en' } },
      route: { redirect: true },
      inlineScripts: {
        routeInfo: {
          script: 'script',
        },
      },
      clientScripts: ['main.js', 'bootstrap.js'],
      config: {},
      plugins: [],
    })

    const bodyWithMeta = renderer
      .create(
        <BodyWithMeta className="body">
          <div>static page</div>
        </BodyWithMeta>
      )
      .toJSON()

    expect(bodyWithMeta).toMatchSnapshot()
  })
})
