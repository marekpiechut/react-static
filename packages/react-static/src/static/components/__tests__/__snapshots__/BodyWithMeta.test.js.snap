// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyWithMeta when route is a redirect route 1`] = `
<body
  className="body"
  lang="en"
>
  <div>
    static page
  </div>
</body>
`;

exports[`BodyWithMeta when route is a static route 1`] = `
<body
  className="body"
  lang="en"
>
  <div>
    static page
  </div>
  <script
    dangerouslySetInnerHTML={
      {
        "__html": "script",
      }
    }
    type="text/javascript"
  />
  <script
    defer={true}
    src="/assets/path/main.js"
    type="text/javascript"
  />
  <script
    defer={true}
    src="/assets/path/bootstrap.js"
    type="text/javascript"
  />
</body>
`;
